### PR TITLE
RUM-5526 Add the integration tests for FeatureSdkCore public API

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/FeatureSdkCoreContextTest.kt [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+            ApplicationProvider.getApplicationContext(),
+            fakeConfigCrashReportsEnabled,
+            fakeTrackingConsent
+        ) as InternalSdkCore
+
+        // When
+        val crashReportsFeature = internalSdkCore.getFeature(CRASH_REPORTS_TASK_NAME)?.unwrap<Feature>()
+
+        // Then
+        checkNotNull(crashReportsFeature)
```
Comment (0xnm) [inline reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/FeatureSdkCoreContextTest.kt:119]: just to be consistent with assertion API usage

```suggestion
       assertThat(crashReportsFeature).isNotNull()
```

File: reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/FeatureSdkCoreContextTest.kt [hunk 2]
```diff
@@ -0,0 +1,638 @@
...
+
+        // Then
+        assertThat(retrievedFeatures).containsAll(registeredFeatures)
+    }
+
+    @Test
+    fun isThreadSafe_when_registerSameFeature_fromDifferentThreads() {
+        // Given
+        val countDownLatch = CountDownLatch(10)
+        val threads = mutableListOf<Thread>()
```
Comment (0xnm) [inline reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/FeatureSdkCoreContextTest.kt:174]: seems it is not used? or you want to call `.join` on each? in this case latch is not needed

File: reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/FeatureSdkCoreContextTest.kt [hunk 3]
```diff
@@ -0,0 +1,638 @@
...
+        countDownLatch.await(SHORT_WAIT_MS, TimeUnit.MILLISECONDS)
+        assertThat(stubFeatureEventReceiver.getReceivedEvents()).containsExactlyInAnyOrder(fakeEvent1, fakeEvent2)
+    }
+
+    // endregion
+
+    // region SingleThreadExecutor Service
+
+    @Test
+    fun mustCreateASingleThreadExecutorService_when_createSingleThreadExecutorService() {
```
Comment (0xnm) [inline reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/FeatureSdkCoreContextTest.kt:602]: executors created in this test and in the next one will be running until test suite finishes the execution, taking some memory. Should we shut them down in the end of the test?

Comment (mariusc83) [reply reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/FeatureSdkCoreContextTest.kt:602]: yup good call

File: reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/FeatureSdkCoreContextTest.kt (hunk 2 above)
Comment (mariusc83) [reply reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/FeatureSdkCoreContextTest.kt:174]: yes was using it before but then switched to countDownLatch

**Pull request merged**